packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -745,9 +745,13 @@ export default (superClass: Class<Parser>): Class<Parser> =>
 
     tsParseFunctionOrConstructorType(
       type: "TSFunctionType" | "TSConstructorType",
+      abstract?: boolean,
     ): N.TsFunctionOrConstructorType {
       const node: N.TsFunctionOrConstructorType = this.startNode();
       if (type === "TSConstructorType") {
+        // $FlowIgnore
+        node.abstract = !!abstract;
+        if (abstract) this.next();
         this.expect(tt._new);
```

Comment:
```suggestion
        this.next(); // eat `new`
```
Since it has been guarded with `this.match(tt._new)` or `this.lookahead().type === tt._new`.